.drone.yml

```diff
@@ -186,6 +201,97 @@ steps:
     depends_on:
       - build for test
 
+---
+kind: pipeline
+type: kubernetes
+name: callisto-ui-prod-deploy
+
+platform:
+  os: linux
+  arch: amd64
+
+depends_on:
+  - callisto-ui-test-deploy
+
+trigger:
+  event:
+    - promote
+  target:
+    include:
+    - production
+    exclude:
+    - test
+
+steps:
+  - name: environment check
+    pull: if-not-exists
+    image: alpine:latest
+    commands:
+      - echo Cannot promote to a non prod environment
+      - exit 1
+    when:
+      target:
+        exclude:
+          - production
+
+  - name: successful build check
+    pull: if-not-exists    
+    image: drone/cli:1.6.2-alpine
+    environment:
+      DRONE_SERVER: https://drone-gh.acp.homeoffice.gov.uk
+      DRONE_TOKEN:
+        from_secret: drone_token        
+    commands:
+    - |        
+      PAGE=1
+      SUCCESSFUL_BUILDS=''      
+      while [ "$PAGE" -le 10 ]; do 
+        SUCCESSFUL_BUILDS="$(drone build ls --format '{{.Number}}' $DRONE_REPO --branch main --status success --event push --limit 100 --page $PAGE)"
+        if [ `echo $SUCCESSFUL_BUILDS | grep -c $DRONE_BUILD_PARENT` -gt 0 ]; then
+          echo 'Found match.'
+          exit 0
+        fi 
+        PAGE="$((PAGE+1))"
+      done
+      echo 'NO match.'
```

Comment:
nit: could we add a tad more detail to this error log as it might help when debugging a failed promotion.